### PR TITLE
feat: add functions for selecting features

### PR DESCRIPTION
**Pull request by eemilkoskinen**
## Description 

Add functions for selecting features

## Type of change

- [ ] Bug fix
- [X] New feature
- [ ] Other

- [X] Non-breaking change
- [ ] Breaking change

## Developer checklist 

- [ ] A [CHANGELOG entry] has been included (only when change is visible to users)

[CHANGELOG entry]: https://github.com/nlsfi/geogen-algorithms/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/geogenalg/selection.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    Returns:
+    -------
+        A filtered GeoDataFrame containing only the polygons that meet the criteria.
+
+    """
+    polygons_intersecting_point = polygon_gdf.geometry.apply(
+        lambda poly: point_gdf.geometry.intersects(poly).any()
+    )
+
+    with_point = polygon_gdf[polygons_intersecting_point]
```
Comment (nlssjritakallio) [inline src/geogenalg/selection.py:93]: It seems that VS Code doesn't recognise "geometry" correctly as a Geoseries in the following lines and it's methods/propertys are not typed. Therefore you could add typings to these two earlier lines so that types and editing work nicely:

    with_point: gpd.GeoDataFrame = ...
    without_point: gpd.GeoDataFrame = ...

File: test/test_selection.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                ],
+                crs="EPSG:3067",
+            ),
+            gpd.GeoDataFrame(
+                geometry=[Polygon([(0, 0), (2, 0), (2, 2), (0, 2)])], crs="EPSG:3067"
+            ),
+            {1, 2},
+        ),
+    ],
+    ids=["one_edge_line"],
```
Comment (nlssjritakallio) [inline test/test_selection.py:135]: Sorry I don't understand this test case id "one_edge_line" - what does it mean? Perhaps you could clarify it. At the moment you have only one test case inside the parametrization but you could also add another one: Test that your function returns expected results if boundary evaluates to None. Because now that case is not covered by unit tests.

**Review by nlssjritakallio — CHANGES_REQUESTED** [review]
Looks good! Few comments and one question/topic that we can discuss tomorrow after daily.

Main branch is now one commit ahead of this branch so for the sake of practise and linear commit history we should also do a rebase before merge. I'll show that too.

File: src/geogenalg/selection.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        )
+        lines_to_keep.append(keep)
+
+    return input_gdf[lines_to_keep]
+
+
+def split_polygons_by_point_intersection(
+    polygon_gdf: gpd.GeoDataFrame,
+    point_gdf: gpd.GeoDataFrame,
+) -> gpd.GeoDataFrame:
```
Comment (nlssjritakallio) [inline src/geogenalg/selection.py:69]: Return value should be tuple[gpd.GeoDataFrame, gpd.GeoDataFrame]

File: test/test_selection.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+                geometry=[LineString([(0, 0), (1, 1)])],
+                crs="EPSG:3067",
+            ),
+            gpd.GeoDataFrame(geometry=[], crs="EPSG:3067"),
+            {0},
+        ),
+    ],
+    ids=["one_line_on_edge_two_not_on_edge", "empty_polygon_boundary_none"],
+)
+def test_remove_lines_on_polygon_edges(
```
Comment (nlssjritakallio) [inline test/test_selection.py:153]: As we spoke on Teams, you could expand these test cases with for example crossing lines and lines with partly on boundary and partly not. Then you could add more than one polygon feature to include union and boundary operations to test cases.

**Review by nlssjritakallio — APPROVED** [review]
